applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp

```diff
@@ -446,6 +458,30 @@ void DistanceModificationProcess::DeactivateFullNegativeElements() {
     }
 }
 
+void DistanceModificationProcess::SetContinuousDistanceToSplitFlag()
+{
+    #pragma omp parallel for
+    for (int i_elem = 0; i_elem < static_cast<int>(mrModelPart.NumberOfElements()); ++i_elem) {
+        auto it_elem = mrModelPart.ElementsBegin() + i_elem;
+        auto &r_geom = it_elem->GetGeometry();
+        std::vector<double> elem_dist;
+        for (unsigned int i_node = 0; i_node < r_geom.PointsNumber(); ++i_node) {
+            elem_dist.push_back(r_geom[i_node].FastGetSolutionStepValue(DISTANCE));
+        }
+        this->SetElementToSplitFlag(*it_elem, elem_dist);
+    }
+}
+
+void DistanceModificationProcess::SetDiscontinuousDistanceToSplitFlag()
+{
+    #pragma omp parallel for
+    for (int i_elem = 0; i_elem < static_cast<int>(mrModelPart.NumberOfElements()); ++i_elem) {
+        auto it_elem = mrModelPart.ElementsBegin() + i_elem;
+        const auto &r_elem_dist = it_elem->GetValue(ELEMENTAL_DISTANCES);
+        this->SetElementToSplitFlag(*it_elem, r_elem_dist);
```

Comment:
you could do `it_elem->GetValue(ELEMENTAL_DISTANCES)` inside the function since you pass the iterator

Reply:
True (I added the iterator after the value). Thanks for noticing.

Reply:
False. In this way the same function can be used for both the continuous (nodal) and discontinuous (elemental) distance cases.